### PR TITLE
Pusher channel subscription failure

### PR DESCRIPTION
**Pull request by hunain-shahid**

### CODE REVIEW

File: Source/API/KUSPushClient.m [hunk 1]
```diff
@@ -103,6 +103,11 @@ - (void)_connectToChannelsIfNecessary
                 _isPusherTrackingStarted = NO;
                 
                 [self _updateStats:^{
+                    // Get latest session on update to avoid packet loss during socket connection
+                    if (_sessionUpdated) {
+                        [_userSession.chatSessionsDataSource fetchLatest];
```
Comment (joeyfromspace) [inline Source/API/KUSPushClient.m]: This is a good addition, but I wonder if we are updating `_lastActivity` when a new message comes in over pusher to make sure we are not fetching if we already have a local copy of the message?

Comment (hunain-shahid) [reply Source/API/KUSPushClient.m]: We are not updating the `_lastActivity` value on pusher update. But as a result, if the user received any message via Pusher then we may have 1 extra call after 60 sec.

**Comment by joeyfromspace** [comment]
### 
We tested out three cases here. There's 2/3 were good. There is one case we need to fix.

- [X] Poll API for messages on failure to subscribe to pusher channel
- [X] On successful pusher connection, continue to poll stats periodically while connected to pusher
- [ ] On successful pusher connection, fallback to API to retrieve messages dropped by pusher

The last case was not successful. We have found that `/c/v1/chat/customers/stats` was still being polled, however messages we were sending from the kustomer app were not being received.

### Steps to reproduce:

1. Return out of the pusher message received listener to simulate pusher not transmitting a message:

screenshot

2. Send a message from Kustomer app to a mobile user

### Expected results:

Message is retrieved on next polling interval

### Actual results:

Stats is polled, however the missing messages are not retrieved from the API

screen

cc @obukspan

**Comment by joeyfromspace** [comment]
Can confirmed that issue is fixed. @robotmonkey have you tested against the rest of the matrix and verified this PR is good to go?

**Pull request merged**